core/version.h

```diff
@@ -3,9 +3,20 @@
  * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
  */
 
+/*
+ * version.h.in is a template file.  version.h is a generated file.
+ * Please do not edit both files directly.
+ *
+ * Any changes to the version should be done in the version.cmake file.
```

Comment:
can you document the procedure?
eg. 1. update version.cmake, 2. run something (what?) to regen this file, 3. commit this generated file as well.